### PR TITLE
fixes for revers proxy in publisher

### PR DESCRIPTION
**Pull request by isharac**
fix for https://github.com/wso2/product-apim/issues/6467 in publisher app  

Removed hard coded /publisher context.
Introduced REVERSE_PROXY config.

const REVERSE_PROXY = {
    enabled: false,
    origin: 'https://localhost:9443',
};
//todo enhance to support REVERSE_PROXY.enabled = "auto" usecase.

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/public/theme/defaultTheme.js [hunk 1]
```diff
@@ -1,4 +1,8 @@
 const APP_CONTEXT = '/publisher';
+const REVERSE_PROXY = {
+    enabled: false,
+    origin: 'https://localhost:9443',
+};
```
Comment (bhathiya) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/public/theme/defaultTheme.js]: @chanaka3d shouldn't these go to the new settings.js?

Comment (chanaka3d) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/public/theme/defaultTheme.js]: Yes.. But the file is not there in the publisher app. Ideally we need publisher and store to have the same structure..

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/services/login/login_callback.jag [hunk 2]
```diff
@@ -68,7 +73,7 @@
     var cookie = {
         'name': 'AM_ACC_TOKEN_DEFAULT_P2',
         'value': accessTokenPart2,
-        'path': site.context + "/",
+        'path': app.context + "/",
```
Comment (malinthaprasan) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/services/login/login_callback.jag]: We shouldn't need to rewrite cookies. The proxy server can do it on behalf of us: Ex:

proxy_cookie_path /publisher /custom-publisher;

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/services/login/login_callback.jag [hunk 3]
```diff
@@ -135,12 +140,12 @@
     if (state !== null) {
         state = decodeURIComponent(state);
         if ( state !== '/') {
-            response.sendRedirect(site.context + state);
+            response.sendRedirect(app.context + state);
```
Comment (malinthaprasan) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/services/login/login_callback.jag]: Same here. A proxy server can also rewrite redirections: Eg: (nginx)

`proxy_redirect https://localhost:9443/publisher https://localhost/custom-publisher;`

Comment (bhathiya) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/services/login/login_callback.jag]: +1, only the callback URL needs to be populated using the origin, as it goes as a query param.

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/services/login/login_callback.jag (hunk 2 above)
Comment (isharac) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/services/login/login_callback.jag]: As offline discussed  with @malinthaprasan , we will address this change later.

**Pull request merged**